### PR TITLE
Add examples

### PR DESCRIPTION
**Pull request by notriddle**
Part of #31: C-EXAMPLE

### CODE REVIEW

**Comment by notriddle** [comment]
Fixed up the PR so all the new doctests actually pass.

File: src/lib.rs [hunk 1]
```diff
@@ -161,6 +224,27 @@ impl<'a> Ammonia<'a> {
     /// Configures a `rel` attribute that will be added on links.
     ///
     /// If `rel` is in the generic or tag attributes, this must be set to `None`.
+    /// Common `rel` values to allow include:
+    ///
+    /// * noopener: This prevents [a particular type of XSS attack],
+    ///   and should usually be turned on for untrusted HTML.
+    /// * noreferrer: This prevents the browser from [sending the source URL]
+    ///   to the website that is linked to.
+    /// * nofollow: This prevents search engines from [using this link for
```
Comment (lnicola) [inline src/lib.rs]: Backticks

File: src/lib.rs [hunk 2]
```diff
@@ -172,25 +256,69 @@ impl<'a> Ammonia<'a> {
...
         self.strip_comments = value;
         self
     }
 
     /// Constructs an `Ammonia` instance configured with the default options.
+    ///
+    /// # Examples
+    ///
+    ///     use ammonia::*;
+    ///     let a = Ammonia::new() // <--
```
Comment (lnicola) [inline src/lib.rs]: Maybe a better example here

File: src/lib.rs [hunk 3]
```diff
@@ -172,25 +256,69 @@ impl<'a> Ammonia<'a> {
...
         Self::default()
     }
 
     /// Sanitizes an HTML fragment in a string according to the configured options.
+    ///
+    /// # Examples
+    ///
+    ///     use ammonia::*;
+    ///     let a = Ammonia::new()
+    ///         .clean("<!-- no -->"); // <--
```
Comment (lnicola) [inline src/lib.rs]: and here.

File: src/lib.rs [hunk 4]
```diff
@@ -135,24 +157,65 @@ impl<'a> Ammonia<'a> {
     /// The value is structured as a map from tag names to a set of attribute names.
     ///
     /// If a tag is not itself whitelisted, adding entries to this map will do nothing.
+    ///
+    /// # Examples
+    ///
+    ///     use ammonia::*;
+    ///     let tags = ["my-tag"].into_iter().cloned().collect();
```
Comment (lnicola) [inline src/lib.rs]: Maybe `hashset!` and friends?

**Comment by lnicola** [comment]
The remaining ones are minor, I suppose, and you might not want them, so I'll approve this pre-emptively.

**Pull request merged**